tests/integration/helpers.py

```diff
@@ -41,8 +42,20 @@ async def loki_rules(ops_test, app_name) -> dict:
         response = urllib.request.urlopen(f"{url}/loki/api/v1/rules", data=None, timeout=2.0)
         if response.code == 200:
             return yaml.safe_load(response.read())
-        return {}
+        return ""
```

Comment:
Why the changed return type when we say it's `dict`?

Reply:
Oops I was supposed to do that change for `loki_endpoint_request`. Fixed.